pkg/cloud/instance.go

```diff
@@ -139,6 +138,37 @@ func (c *client) ResolveTemplate(
 	return templateID, nil
 }
 
+func (c *client) ResolveDiskOffering(csMachine *infrav1.CloudStackMachine) (diskOfferingID string, retErr error) {
+	if len(csMachine.Spec.DiskOffering.ID) > 0 {
+		csDiskOffering, count, err := c.cs.DiskOffering.GetDiskOfferingByID(csMachine.Spec.DiskOffering.ID)
+		if err != nil {
+			return "", multierror.Append(retErr, errors.Wrapf(
+				err, "could not get DiskOffering by ID %s", csMachine.Spec.DiskOffering.ID))
+		} else if count != 1 {
+			return "", multierror.Append(retErr, errors.Errorf(
+				"expected 1 DiskOffering with UUID %s, but got %d", csMachine.Spec.DiskOffering.ID, count))
+		}
+
+		if len(csMachine.Spec.DiskOffering.Name) > 0 && csMachine.Spec.DiskOffering.Name != csDiskOffering.Name {
+			return "", multierror.Append(retErr, errors.Errorf(
+				"diskOffering name %s does not match name %s returned using UUID %s", csMachine.Spec.DiskOffering.Name, csDiskOffering.Name, csMachine.Spec.DiskOffering.ID))
```

Comment:
We don't have a line length enforcing linter in place yet, but could you please add a or many newlines to this one to make it a bit easier to read?